### ISSUE TITLE
Instapy skipping my DESIRED business categories?

### ISSUE THREAD

**Issue by flygohr** [InstaPy/InstaPy#5011]
## Expected Behavior
InstaPy should allow users in my current, desired business categories:

```
session.set_skip_users(skip_private=True,
                           skip_no_profile_pic=True,
                           skip_business=True,
                           dont_skip_business_categories=['Advertising Agency', 'Advertising/Marketing', 'Art', 'Art Gallery', 'Art Museum', 'Artist', 'Arts & Entertainment', 'Arts & Humanities Website', 'Clothing Store', 'Content & Apps', 'Creators & Celebrities', 'Graphic Designer', 'Media/News Company', 'Personal Goods & General Merchandise Stores', 'Photographer', 'Public Figure', 'Publishers'])
```

## Current Behavior
Instapy is currently skipping them instead??

```
INFO [2019-09-05 15:20:37] [...]  Ongoing Follow [6/30]: now following 'daniedoesart'...
INFO [2019-09-05 15:20:41] [...]  User: 'daniedoesart'  |> followers: 171  |> following: 144  |> relationship ratio: 1.18
INFO [2019-09-05 15:20:41] [...]  'daniedoesart' has a business account in the undesired category of 'Creators & Celebrities'
```

```
INFO [2019-09-05 15:25:19] [...]  Ongoing Follow [15/30]: now following 'lazura_katalog'...
INFO [2019-09-05 15:25:23] [...]  User: 'lazura_katalog'  |> followers: 35  |> following: 32  |> relationship ratio: 1.09
INFO [2019-09-05 15:25:23] [...]  'lazura_katalog' has a business account in the undesired category of 'Personal Goods & General Merchandise Stores'
```

## InstaPy configuration
```
# Get a session!
session = InstaPy(username=insta_username,
                  password=insta_password,
                  headless_browser=True,
                  disable_image_load=True,
                  nogui=True,
                  multi_logs=True)

				  
# SCRIPT START

with smart_run(session):
    # General settings
    session.set_dont_include(friends)
    session.set_dont_like(dont_likes)
    session.set_ignore_if_contains(ignore_list)
    session.set_ignore_users(ignore_users)
    session.set_simulation(enabled=True)
    session.set_relationship_bounds(enabled=True,
                                    potency_ratio=None,
                                    delimit_by_numbers=True,
                                    max_followers=100000,
                                    max_following=3000,
                                    min_followers=100,
                                    min_following=25,
                                    min_posts=10)

    session.set_skip_users(skip_private=True,
                           skip_no_profile_pic=True,
                           skip_business=True,
                           dont_skip_business_categories=[target_bc])

    session.set_user_interact(amount=3, randomize=True, percentage=80, media='Photo')
    session.set_do_like(enabled=True, percentage=50)
    session.set_do_follow(enabled=True, percentage=75, times=1)

    # ACTIVITIES

    # FOLLOW+INTERACTION on TARGETED accounts
    # Select users form a list of a predefined targets...
    number = random.randint(3, 5)
    random_targets = targets

    if len(targets) <= number:
        random_targets = targets

    else:
        random_targets = random.sample(targets, number)

    # INTERACT with the chosen targets...
    session.follow_user_followers(random_targets,
                                  amount=follow_per_session,
                                  randomize=True, sleep_delay=600,
                                  interact=True)

    # UNFOLLOW nonfollowers after one day...
    session.unfollow_users(amount=unfollow_per_session, instapy_followed_enabled=True, instapy_followed_param="nonfollowers", style="FIFO", unfollow_after=24*60*60, sleep_delay=501)

    # UNFOLLOW all users followed by InstaPy after one week to keep the following-level clean...
    session.unfollow_users(amount=unfollow_per_session, instapy_followed_enabled=True, instapy_followed_param="all", style="FIFO", unfollow_after=198*60*60, sleep_delay=501)
```

**Comment by realsnad** [comment]
There is a bug in the code. Incorrect parameter skip_bio_keyword position in call to validate_username in validate_user_call in instapy.py.

To fix this you need to move "self.skip_bio_keyword," just before "self.logger," in call to validate_username inside validate_user_call.

If you're using 0.6.3 then it would be switching lines 1321 and 1322 in instapy.py.

So basically, this bug breaks both skip_bio and skip_business_categories. This needs to be fixed in the next version.

**Comment by flygohr** [comment]
Trying this rn

**Comment by flygohr** [comment]
Oooook, now I get only '-empty list' and 'Sorry, couldn't check for number of posts of username' :/

**Comment by realsnad** [comment]
And now you can go ahead and read my issue #5042.

That’s what is happening with your profile right now.

**Comment by flygohr** [comment]
Oooook, I managed to get my script working again BUT

Even if those lines are as you described (it's been fixed in 0.6.4 too) 

I am still getting that error, instapy is skipping users in my desired business categories

I tried escaping the '&' character in some names (as I see them mostly being skipped) but nothing changed
